### ISSUE TITLE
Genesis/MD - games built for original 3 button controller go straight to "Game Over" 

### ISSUE THREAD

**Issue by calburitel** [jasarien/Provenance#376]
Certain Genesis games won't work with a 6-button controller "attached" (Sega came out with this controller scheme later in the life of the console), and this is something that's supposed to be taken care of by emulation of the "mode" button, but I don't believe the "mode" button is mapped anywhere on Provenance. It could not be across the board, but I noticed this with Forgotten Worlds. Most games work fine, but not Forgotten Worlds. Instead it will give you the Game Over screen when you start the game.

This seemed to be a common issue with OpenEmu and other Genesis/MD emulators. Not sure if this is helpful (pardon my ignorance), but here's the link to that fix in OpenEmu. Would it be easier to add a 3 button scheme option?

https://github.com/OpenEmu/GenesisPlus-Core/commit/074c0207a151d3f7bed14aa40d10be5a9f5f4f25

Provenance: Version 1.2.7
Forgotten Worlds (World) (v1.1).md
iPad Pro 9.7 Model MLQ62LL/A
iOS 9.3.2

**Comment by jasarien** [comment]
Ah, weird issue. I'll look into a fix as soon as I get a spare moment.

**Comment by calburitel** [comment]
Thanks! Totally not a priority in the grand scheme of things. Thank you for everything that you do. I can't tell you how much I love having Provenance on my iPad.

**Comment by sevdestruct** [comment]
Confirming this is still a bug present in recent swift based dev branches... @jasarien Didn't you mention there is a way to detect which games didn't require 6x button controller and defer to 3x -- woukpld thst resolve this?

**Comment by jasarien** [comment]
Yeah, I’m not sure if it’s built into the emulation core wer’re using though (GenesisPluxGX). If it is, then it’s just a case of responding to whatever callback or event the emulator sends when this case occurs and switching the controller type the core has “plugged in”.

**Comment by JoeMatt** [comment]
If you look at GenPlusGameCore.m in OpenEMU they have a way of detecting by looking at the game ROM header and also an override list pad6Buttons for broken roms. But I think that's only an override for games that GenXPlus can't detect on its own. It isn't detecting for us because the core needs an update. 1.7.5 changelog states, 'added gamepad type auto-detection'. I already added #736 for this update.
